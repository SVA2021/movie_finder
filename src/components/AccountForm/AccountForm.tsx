import {Box, Button, Input} from '@mui/material';
import {FC, memo} from 'react';
import {Controller, SubmitHandler, useForm} from 'react-hook-form';
import {TUser} from '../../features/auth/authSlice';

interface AccountFormProps {
  accountSubmit: (data: TUser) => void
}

export const AccountForm: FC<AccountFormProps> = memo(({accountSubmit}) => {

  const {control, handleSubmit} = useForm<TUser>();
  const onSubmit: SubmitHandler<TUser> = data => accountSubmit(data);

  return (
    <Box
      data-testid={'account-form'}
      component={'form'}
      onSubmit={handleSubmit(onSubmit)}
    >
      <Controller
        name={'login'}
        control={control}
        defaultValue={''}
        render={({field}) => (
          <Input
            sx={{
              m: 1,
              p: 1,
              bgcolor: 'secondary.main',
              color: 'primary.main',
              borderRadius: 2,
              opacity: 0.5,
              '&:hover': {
                opacity: 0.75,
              },
              '&:focus-within': {
                opacity: 1,
              },
            }}
            {...field}
            fullWidth
            required
            placeholder='логин'
          />
        )}
      />
      <Controller
        name={'password'}
        control={control}
        defaultValue={''}
        render={({field}) => (
          <Input
            sx={{
              m: 1,
              p: 1,
              bgcolor: 'secondary.main',
              color: 'primary.main',
              borderRadius: 2,
              opacity: 0.5,
              '&:hover': {
                opacity: 0.75,
              },
              '&:focus-within': {
                opacity: 1,
              },
            }}
            {...field}
            fullWidth
            required
            placeholder='пароль'
          />
        )}
      />
      <Box textAlign={'center'} mt={2} >
        <Button type={'submit'} >Войти</Button>
      </Box>
    </Box>
  );
});