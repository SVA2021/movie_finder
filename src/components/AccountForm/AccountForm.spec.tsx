import {fireEvent, render, screen, waitFor} from "@testing-library/react";
import {AccountForm} from "./AccountForm";

const mockLogin = jest.fn(({login, password}) => {
  return Promise.resolve({login, password});
});

describe('render AccountForm', () => {

  const setUp = () => render(<AccountForm accountSubmit={({login, password}) => mockLogin({login, password})} />);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('account-form')).toMatchSnapshot();
  });

  it('check form elements is rendered', () => {
    setUp();
    expect(screen.getByPlaceholderText(/логин/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/пароль/i)).toBeInTheDocument();
    expect(screen.getByText(/Войти/i)).toBeInTheDocument();
  });

  it('check form can input values', () => {
    setUp();
    fireEvent.change(screen.getByPlaceholderText(/логин/i), {target: {value: "login"}})
    expect(screen.getByPlaceholderText(/логин/i)).toHaveValue('login');
    fireEvent.change(screen.getByPlaceholderText(/пароль/i), {target: {value: "password"}})
    expect(screen.getByPlaceholderText(/пароль/i)).toHaveValue('password');
  });

  // it('check form can be submitted', async () => {
  //   setUp();

  //   fireEvent.input(screen.getByPlaceholderText(/логин/i), {target: {value: "login"}})
  //   fireEvent.input(screen.getByPlaceholderText(/пароль/i), {target: {value: "password"}})
  //   fireEvent.submit(screen.getByRole('button'))

  //   await waitFor(() => expect(screen.queryAllByRole("button")).toHaveLength(1));
  //   expect(mockLogin).toBeCalledWith("login", "password");
  // });
})