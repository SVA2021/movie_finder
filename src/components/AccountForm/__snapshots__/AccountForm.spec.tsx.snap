// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AccountForm snapshot test 1`] = `
<form
  class="MuiBox-root css-0"
  data-testid="account-form"
>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-acv73g-MuiInputBase-root-MuiInput-root"
    required=""
  >
    <input
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      name="login"
      placeholder="логин"
      required=""
      type="text"
      value=""
    />
  </div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-acv73g-MuiInputBase-root-MuiInput-root"
    required=""
  >
    <input
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      name="password"
      placeholder="пароль"
      required=""
      type="text"
      value=""
    />
  </div>
  <div
    class="MuiBox-root css-16j30i3"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Войти
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</form>
`;
